### PR TITLE
ddl: query since min job id

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #54436

Problem Summary:

### What changed and how does it work?
second pr mentioned in https://github.com/pingcap/tidb/pull/54547/, now we pessimistic lock global id, and generate global id and insert job in one txn, so we can make sure when job of ID A which is the minimum in a txn submitted, no jobs of ID < A can be inserted, so it's safe to query since min job id
- query since min job id to mitigate slowness due to un-compacted deleted kv, see this issue https://github.com/pingcap/tidb/issues/52905. by query since min job id, tikv can directly seek to the starting KV and skip all those that have deleted but not compacted without scanning

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

same case as in https://github.com/pingcap/tidb/pull/54581, takes 11m24, and there is very little speed degradation now

image

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/ddl_worker.go [hunk 1]
```diff
@@ -411,16 +411,17 @@ func (d *ddl) addBatchDDLJobs(jobWs []*JobWrapper) error {
 		return nil
 	}
 
+	ctx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnDDL)
```
Comment (lance6716) [inline pkg/ddl/ddl_worker.go:414]:
```suggestion
	ctx := kv.WithInternalSourceType(d.ctx, kv.InternalTxnDDL)
```

File: pkg/ddl/job_table.go [hunk 2]
```diff
@@ -97,12 +97,14 @@ var _ owner.Listener = (*ownerListener)(nil)
...
 	l.scheduler = &jobScheduler{
-		schCtx:       ctx,
-		cancel:       cancelFunc,
-		runningJobs:  newRunningJobs(),
-		sysTblMgr:    systable.NewManager(l.ddl.sessPool),
-		schemaLoader: l.ddl.schemaLoader,
+		schCtx:            ctx,
+		cancel:            cancelFunc,
+		runningJobs:       newRunningJobs(),
+		sysTblMgr:         sysTblMgr,
```
Comment (lance6716) [inline pkg/ddl/job_table.go:105]: can we use ddl.sysTblMgr?

**Review by lance6716 — APPROVED** [review]

File: pkg/ddl/job_table.go (hunk 2 above)
Comment (D3Hunter) [reply pkg/ddl/job_table.go:105]: it's just a wrapper around session pool, both lgtm

**Review by tangenta — APPROVED** [review]

**Pull request merged**